scrapy_autounit/utils.py

```diff
@@ -170,11 +191,17 @@ def test(self):
         with open(fixture_path) as f:
             data = json.load(f)
         fixture_items = data['items']
-        response = HtmlResponse(encoding='utf-8', **data['response'])
+
+        data['request'].pop('_encoding', None)
+        data['request'].pop('callback', None)
+
+        request = Request(callback=callback, **data['request'])
+        response = HtmlResponse(encoding='utf-8', request=request, **data['response'])
+
         callback_items = list(callback(response))
         for index, item in enumerate(callback_items):
             fixture_data = fixture_items[index]['data']
             if isinstance(item, Request):
                 item = parse_request(item, spider=spider)
-            self.assertEqual(fixture_data, item, 'Not equal!')
+            self.assertEqual(fixture_data, dict(item), 'Not equal!')
```

Comment:
So might be a little late, but is assertEqual output helpful for large diffs?  I think we might be using https://pypi.org/project/datadiff/ somewhere which is minorly easier to read.

Reply:
It's actually useful if you set `maxDiff = None` so you can see the full diff and you're comparing dicts like this case (it uses `assertDictEqual` internally). The output is understandable I think, anyway we could use datadiff if it's better...

Reply:
Yeah, not important, just dumping an idea.  I thought the default diff just printed both of them and says "they're different" rather than highlighting the actual fields that mismatched or something like that.  Python 3.5+ might have improved things though.